src/te/operation/cross_thread_reduction.cc

```diff
@@ -48,9 +97,18 @@ Stmt MakeCrossThreadReduction(const ComputeOpNode* self, const Stage& stage,
     CHECK(reduce);
     reduces[i] = reduce;
   }
-  PrimExpr cond = reduces[0]->condition;
-  for (PrimExpr v : conds) {
-    cond = cond && v;
+
+  // This computes the bound checking predicates in normal reduction.
+  auto normal_preds =
+      MakeBoundCheck(stage, dom_map, value_map, false, std::unordered_set<IterVar>());
+
+  // The existing reduction predicate (only from the first one one?)
+  PrimExpr input_pred = reduces[0]->condition;
+
+  // normal_pred = input_pred && normal_pred
+  normal_preds.push_back(input_pred);
+  for (PrimExpr v : normal_preds) {
+    if (v.defined()) normal_preds.push_back(v);
   }
```

Comment:
What does this loop do? Iterating through `normal_preds` and push into itself?

Reply:
Yes, this looks odd. I saw null predicates in this vector. So this is to remove them. Let me add a comment.

Reply:
Thanks for catching this! normal_preds may contain null  expressions, so we need to filter them out.  I just fixed it.